### PR TITLE
fix(core): fix pickling of undefined functions

### LINKED ISSUE #26873: Saving a long expression to load in another python notebook/file
I have multiple expressions that are huge. 

I make a list to easily handle them: 

`eqnslist = [eqn0, eqn1, eqn3, eqn4]`

To calculate these equations (expressions) it takes about 5-10 minutes. I would rather be able to save them into a file and load them the next time I run the notebook. 

I've tried pickle and json and both make a similar error like: 

`Can't pickle attribute: `

So what's the best way to save sympy expressions/variables?

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### Brief description of what is fixed or changed

Fixes gh-26873

#### Release Notes

* utilities
    * It is now possible to pickle expressions containing undefined functions (like `Function('f')`). Previously trying to pickle such expressions would fail with an exception.

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This is ready for review.

**Comment by oscarbenjamin** [comment]
This one is still ready for review.

It has in fact already been merged in diofant.

File: sympy/core/function.py [hunk 1]
```diff
@@ -926,6 +927,20 @@ def _diff_wrt(self):
         return False
 
 
+# Using copyreg is the only way to make a dyanmically generated instance of a
```
Comment (smichr) [inline sympy/core/function.py:930]:
```suggestion
# Using copyreg is the only way to make a dynamically generated instance of a
```

**Comment by jherkenhoff** [comment]
I would love to see this PR in main :+1: :grin:

**Comment by oscarbenjamin** [comment]
> I would love to see this PR in main 👍 😁

Yes, I'm not sure why it is taking so long to get a review for this.

**Comment by oscarbenjamin** [comment]
If someone wants to review this then go ahead. Otherwise I'll merge it soon.

**Pull request merged**